x-pack/plugins/apm/server/index.ts

```diff
@@ -52,7 +51,17 @@ const configSchema = schema.object({
 
 // plugin config
 export const config: PluginConfigDescriptor<APMXPackConfig> = {
-  deprecations: ({ deprecate }) => [deprecate('enabled', '8.0.0')],
+  deprecations: ({ deprecate, renameFromRoot }) => [
+    deprecate('enabled', '8.0.0'),
+    renameFromRoot(
+      'xpack.apm.maxServiceEnvironments',
+      `uiSettings.overrides[${maxSuggestions}]`
```

Comment:
What does this actually do? It looks like we are not using `xpack.apm.maxServiceEnvironments` anywhere in the ui anymore.

Reply:
nvm makes sense now that we are basically renaming the config setting but allowing users to keep using the old name.